src/test/ui/consts/const_fn_mut_refs.rs

```diff
@@ -0,0 +1,15 @@
+// run-pass
+
+#![feature(const_fn_mut_refs)]
+
+struct Foo {
+    x: i32
+}
+
+const fn bar(foo: &mut Foo) -> i32 {
```

Comment:
Please add some tests calling this function in const contexts

Reply:
it seems I need to change more checks:
```
error[E0017]: references in constant functions may only refer to immutable values
  --> /home/christian/Workspace/contrib/rust/src/test/ui/consts/const_fn_mut_refs.rs:14:9
   |
LL |     bar(&mut Foo{x: 0}) - 1
   |         ^^^^^^^^^^^^^^ constant functions require immutable values

error: aborting due to previous error

For more information about this error, try `rustc --explain E0017`.
```

Reply:
I tried to allow mutable borrows in constant bodies via `https://github.com/rust-lang/rust/pull/66606/commits/8c17a13ac0f79b934bf55493335ca874cf4ae041` but apparently I broke `unleash-the-miri-inside-of-you` tests. A lot of warnings are gone now and that doesn't sound like a good thing.

Reply:
These tests should now not be unleash tests anymore but just have the feature gate enabled. Adjust their errors accordingly

Reply:
should I move them to the `const-mut-refs` folder?

Reply:
also the following test
```rust
// failure-status: 101
// rustc-env:RUST_BACKTRACE=0
// normalize-stderr-test "note: rustc 1.* running on .*" -> "note: rustc VERSION running on TARGET"
// normalize-stderr-test "note: compiler flags: .*" -> "note: compiler flags: FLAGS"
// normalize-stderr-test "interpret/intern.rs:[0-9]*:[0-9]*" -> "interpret/intern.rs:LL:CC"

#![feature(const_mut_refs)]
#![allow(const_err)]

use std::cell::UnsafeCell;

// this test ICEs to ensure that our mutability story is sound

struct Meh {
    x: &'static UnsafeCell<i32>,
}

unsafe impl Sync for Meh {}

// the following will never be ok!
const MUH: Meh = Meh {
    x: &UnsafeCell::new(42),
};

fn main() {
    unsafe {
        *MUH.x.get() = 99;
    }
}
```
causes an ICE:
```
thread 'rustc' panicked at 'assertion failed: `(left != right)`
  left: `Const`,
 right: `Const`: UnsafeCells are not allowed behind references in constants. This should have been prevented statically by const qualification. If this were allowed one would be able to change a constant at one use site and other use sites could observe that mutation.', src/librustc_mir/interpret/intern.rs:174:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```

Reply:
The test should keep iceing. If the comments are unclear, we should expand on them.  I'm surprised your feature gate has an effect, without unleash this cell should be statically denied

Reply:
Oh ok, I think the comments are clear, Is just me that got the wording wrong